game-core/src/main/java/games/strategy/engine/data/ProductionRule.java

```diff
@@ -5,6 +5,9 @@
 import games.strategy.triplea.Constants;
 import games.strategy.util.IntegerMap;
 
+/**
+ * A production rule.
```

Comment:
Not much value-add on this one, restates the class name. Describing what a production rule is would be of help, otherwise I would suggest to remove this comment to avoid the maintenance cost